### PR TITLE
Additional fixes we have made for supporting Windows, Mac OS and RTOS32 builds

### PR DESCRIPTION
**Pull request by pm-ur**
### Checklist

* [ ] documentation is added or updated
* [ ] tests are added or updated

### CODE REVIEW

File: Config.cmake.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+include("${CMAKE_CURRENT_LIST_DIR}/@targets_export_name@.cmake")
```
Comment (sjaeckel) [inline Config.cmake.in:1]: There's a huge discussion in #391 around whether cmake should be supported or something else.
One thing is for sure, this cmake solution will not be merged as it's unstructured and incomplete.

File: libtomcrypt_VS2013.sln [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿
```
Comment (sjaeckel) [inline libtomcrypt_VS2013.sln:1]: We decided that we don't add VS projects for all the various versions. There's a 2008 Version and if you want to use it in a newer version, feel free to convert it yourself. _But_ who wants to use a VS project anyways when there's the msvc makefile?

File: src/headers/tomcrypt_custom.h [hunk 3]
```diff
@@ -10,6 +10,26 @@
 #ifndef TOMCRYPT_CUSTOM_H_
 #define TOMCRYPT_CUSTOM_H_
 
+
```
Comment (sjaeckel) [inline src/headers/tomcrypt_custom.h:13]: That's your configuration and shouldn't be upstream.

File: src/misc/crypt/crypt_ltc_mp_descriptor.c [hunk 4]
```diff
@@ -8,9 +8,20 @@
  */
 #include "tomcrypt_private.h"
 
+
 /* Initialize ltc_mp to nulls, to force allocation on all platforms, including macOS. */
-ltc_math_descriptor ltc_mp = { 0 };
+/* ltc_math_descriptor ltc_mp = { 0 }; */
```
Comment (sjaeckel) [inline src/misc/crypt/crypt_ltc_mp_descriptor.c:13]: This is the correct solution according to the C standard and if your compiler is broken we can't help you. (If you want to support broken compilers, please add this code to your fork, but we won't add it.)

**Comment by sjaeckel** [comment]
Thanks for this PR but I will close it as after review IMO nothing remains that should be merged.

The remaining patches I didn't comment on are AFAIU either NOOP's (e.g. the changes in `.gitignore` or the makefiles) or caused by your preference of coding/style/build-system and therefor should also not be integrated here.

Please re-open if you think I've missed something that should be merged.

**Comment by pm-ur** [comment]
@sjaeckel There may be some commits/fixes other than the ones with issues you described which could be useful. 

Just that I didn't have time to separate them specifically for the upstream fork for now, so I created one with all the changes we had in our fork. Maybe I can later create separate PRs for those individually.

**Comment by sjaeckel** [comment]
Feel free to open another PR.

If possible I'd like to have one PR, with the patches topic-wise in single commits and rebased on top of develop :-)

**Pull request closed without merging**